File: src/main/scala/com/bytebreakstudios/tictactoe/scala/action/ChangeTurnAction.scala
package com.bytebreakstudios.tictactoe.scala.action

import com.bytebreakstudios.board.Action
import com.bytebreakstudios.tictactoe.scala.State

class ChangeTurnAction extends Action[State] {
  override def innerApply(t: State): State = t.copy(xTurn = !t.xTurn)

  override def innerUnapply(t: State): State = innerApply(t)

  override def description(): String = ???

  override def serialize(): String = ???

  override def deserialize(s: String) = ???
}

File: src/main/java/com/bytebreakstudios/tictactoe/action/ChangeTurnAction.java
package com.bytebreakstudios.tictactoe.action;

import com.bytebreakstudios.tictactoe.State;
import com.bytebreakstudios.board.Action;

public class ChangeTurnAction extends Action<State> {

    @Override
    protected State innerApply(State state) {
        State next = (State) state.clone();
        next.xTurn = !next.xTurn;
        return next;
    }

    @Override
    protected State innerUnapply(State state) {
        return innerApply(state);
    }

    @Override
    public String description() {
        return null;
    }

    @Override
    public String serialize() {
        return null;
    }

    @Override
    public Object deserialize(String s) {
        return null;
    }
}
